analysis/blacklist/urls.go

```diff
@@ -115,19 +118,77 @@ func fillBlacklistedURL(blURL *data.BlacklistedURL, longURL, db,
 	}
 	blURL.Host = host
 	blURL.Resource = resource
-
-	connQuery := bson.M{"dst": bson.M{"$in": blURLFull.IPs}}
+	httpPipeline := []bson.D{
+		{
+			{ "$match", bson.M{"host": host, "uri": resource} },
+		},
+		{
+			{ "$group", bson.M{
+					"_id": bson.D{
+						{ "h", "$host" },
+						{ "u", "$uri" },
+						{ "s", "$id_origin_h" },
+						{ "d", "$id_resp_h" },
```

Comment:
Unfortunately, your solution still grabs connections which may not have been part of the url request, due to reliance on uconn. 

This may be more correct (and possibly faster) to do in parts.
Start a pipeline on http
match on host and uri
project down to just the UID field
lookup from conn on UID
Project the orig_bytes, resp_bytes, and duration fields out of the conn object into the top level and drop the conn object.
From there, run a group on null (to group all the documents). Then run sum and avg commands similar to these: https://github.com/ocmdev/rita/blob/master/analysis/structure/uconn.go#L61

Maybe a better way to do it

Reply:
Fixed!